### PR TITLE
UX: Enhance Resharing dialogs are respecting resharing permissions

### LINKED ISSUE #1176: UX: Enhance sharing when resharing is forbidden
### Actual behaviour
- error after sharing

### Expected behaviour
- show a text "not allowed to reshare files/folders" instead of whole stuff beneath the horizontal border

 
### Steps to reproduce
1. disable "allow resharing" in server setting
2. try to share a shared file

PR: #1792

### PR DESCRIPTION
**Pull request by vnidens**
resolves #1176 and #1177

### CODE REVIEW

**Comment by mario** [comment]
Thanks for the contribution. @AndyScherzinger  & @tobiasKaminsky can you please review?

**Comment by AndyScherzinger** [comment]
Will do when I am back from vacation 👍

File: src/main/java/com/owncloud/android/datamodel/OCFile.java [hunk 1]
```diff
@@ -687,4 +688,9 @@ public boolean isSharedWithMe() {
         return (permissions != null && permissions.contains(PERMISSION_SHARED_WITH_ME));
     }
 
+    public boolean canReshare(){
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/datamodel/OCFile.java]: Only minor code formatting: please add a whitespace between () and {
(same will we on other occurences here: you can easily do this with androidstudio and Code -> Reformat Code)

File: src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java [hunk 2]
```diff
@@ -144,6 +152,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
+    /**
+     * Get known server capabilities from DB
+     * <p/>
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java]: missing opening < on p tag

File: src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java [hunk 3]
```diff
@@ -144,6 +152,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+     * Depends on the parent Activity provides a {@link com.owncloud.android.datamodel.FileDataStorageManager}
+     * instance ready to use. If not ready, does nothing.
+     */
+    public void refreshCapabilitiesFromDB() {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java]: As this function is only used once I would vote for putting this in onCreate
(also the name is somehow misleading as refresh usually implies "updating infos from server", at least for me).

You can create a new instance of storage manager like this:
FileDataStorageManager storageManager = new FileDataStorageManager(account, getContext().getContentResolver());

This way you can omit the != null checks here and later.

File: src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java (hunk 2 above)
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/fragment/EditShareFragment.java]: fixed via https://github.com/nextcloud/android/pull/1247/commits/a1f60061cdab09796b1b8e41f949f783c622a8e3

File: src/main/java/com/owncloud/android/datamodel/OCFile.java (hunk 1 above)
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/datamodel/OCFile.java]: fixed via https://github.com/nextcloud/android/pull/1247/commits/a1f60061cdab09796b1b8e41f949f783c622a8e3

**Comment by mario** [comment]
@AndyScherzinger @tobiasKaminsky can one of you fix required stuff, so we can merge this?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky what needs to be done here (can you do it?)? because if this one is ready I can also rebase it to master, test and merge...

**Comment by tobiasKaminsky** [comment]
Changes done :+1:

**Comment by AndyScherzinger** [comment]
rebased to latest master, let's see if lint is happy now

**Comment by AndyScherzinger** [comment]
👍 fine codewise, rebase remaining solved lint issue

great work @vnidens 🥇 

merged to master for 1.5.0

**Pull request merged**